mail/mail/doctype/mail_account_request/mail_account_request.py

```diff
@@ -47,15 +57,26 @@ def validate_role(self) -> None:
 		if self.role not in ["Mail User", "Mail Admin"]:
 			frappe.throw(_("Invalid role. Please select a valid role."))
 
-	def validate_is_invite(self) -> None:
-		"""Validates the is_invite field."""
+	def validate_email(self) -> None:
+		"""Validates email if needed."""
 
-		if not self.is_invite:
-			self.role = "Mail Admin"
-			self.invited_by = None
-			self.tenant = None
-			self.domain_name = None
-			self.account = None
+		if self.send_email and not self.email:
+			frappe.throw(_("Email is required to send invite"))
+
+		self.email = self.email.strip().lower()
+		validate_email_address(self.email, True)
+
+	def validate_non_invite(self) -> None:
+		"""Validates self sign up."""
+
+		if frappe.db.exists("User", self.email):
```

Comment:
User.name and User.email can be different after https://github.com/frappe/frappe/issues/26189

```suggestion
		if frappe.db.exists("User", {"email": self.email}):
```